test/types.ts

```diff
@@ -95,3 +95,11 @@ const overloadedResult1Valid: string | void = valid3.overloaded(1);
 // @ts-expect-error overloaded with two arguments returns only void
 const overloadedResult2Invalid: string = valid3.overloaded('string', 1);
 const overloadedResult2Valid: string | void = valid3.overloaded('string', 1);
+
+// A state machine that uses symbols as a state keys
+const valid4 = fsm(Symbol.for('foo'), {
+  [Symbol.for('foo')]: {
```

Comment:
@kenkunz I'm not sure if this state machine will actually work when running it. Have you tested it?
Symbols are unique even if you construct them from the same parameter `foo`.
`Symbol.for('foo') !== Symbol.for('foo')`

Reply:
you would need to store the symbol in a variable and then use the variable wherever you need it

Reply:
`Symbol.for()` – unlike `Symbol()` – uses a global `symbol` registry and always returns the same `symbol` for a given key (see [Symbol.for()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/for)):
```javascript
Symbol('foo') === Symbol('foo')         // false
Symbol.for('foo') === Symbol.for('foo') // true
```
There's actually a bug in the code right now that prevents the use of `symbols` as state keys (dynamic type-check that only allows `string` transitions). I have a fix on a local branch that will be included in `v1.1.0` release.

I don't personally ever expect to use `symbols` as state keys… but I think they should be fully supported, since `strings` and `symbols` are the two valid types of `object` keys. Someone else may have a preference or use case that I'm not thinking of.

Reply:
ah cool, I didn't know that 👍